components/channel_members_dropdown/channel_members_dropdown.jsx

```diff
@@ -123,12 +123,12 @@ export default class ChannelMembersDropdown extends React.Component {
             return null;
         }
 
-        if (this.props.canChangeMemberRoles) {
+        if (this.props.canChangeMemberRoles || this.props.canRemoveMember) {
             const role = this.renderRole(isChannelAdmin, isGuest);
 
             const canRemoveFromChannel = this.props.canRemoveMember && (this.props.channel.name !== Constants.DEFAULT_CHANNEL || isGuest) && !this.props.channel.group_constrained;
-            const canMakeChannelMember = Boolean(isChannelAdmin && !isGuest);
-            const canMakeChannelAdmin = Boolean(supportsChannelAdmin && !isChannelAdmin && !isGuest);
+            const canMakeChannelMember = Boolean(isChannelAdmin && !isGuest && this.props.canChangeMemberRoles);
+            const canMakeChannelAdmin = Boolean(supportsChannelAdmin && !isChannelAdmin && !isGuest && this.props.canChangeMemberRoles);
```

Comment:
So because there's an OR above, you have to re-add the test for canChangeMemberRoles?
Ok, I think I see. 

... Sheesh. Can you make a truth table for this Martin?  :)

Reply:
I was in rush-to-fix-the-bug mode but I'm sure this can be simplified.

Reply:
This whole stack of ifs needs to be simplified, tbh. This PR probably isn't the place to do it.